example/mapreduce/mapreduce_bootstrap_controller.go

```diff
@@ -0,0 +1,146 @@
+package mapreduce
+
+// user_interface provide user a interface to start their mapreduce application
+// MapreduceBootstrapController would controll the bootstrap of controller, task
+// (serves as same as regression/demo/run_regression.sh).
+// It likes a local version of Kubernetes, managing the whole framework status.
+// For user starting a new mapreduce framework, only need to do is
+// invoke NewMapreduceBootstrapController() to get a new controlle,
+// invoke its Start interface with his own mapreduce configuration(represents as map[string]interface{})
+// therefore the mapreduce framework will run automatically.
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+	"github.com/coreos/go-etcd/etcd"
+	"github.com/taskgraph/taskgraph"
+	"github.com/taskgraph/taskgraph/controller"
+	"github.com/taskgraph/taskgraph/framework"
+)
+
+type MapreduceBootstrapController struct {
```

Comment:
@Hongchao,

Can you suggest an name so that we have an example to learn from? Thanks.

Xiaoyun

On Thu, May 7, 2015 at 2:23 PM, Hongchao Deng <notifications@github.com>
wrote:

> In example/mapreduce/mapreduce_bootstrap_controller.go
> <https://github.com/taskgraph/taskgraph/pull/109#discussion_r29827319>:
>
> > +// invoke its Start interface with his own mapreduce configuration(represents as map[string]interface{})
> > +// therefore the mapreduce framework will run automatically.
> > +
> > +import (
> > +	"fmt"
> > +	"log"
> > +	"net"
> > +	"os"
> > +
> > +	"github.com/coreos/go-etcd/etcd"
> > +	"github.com/taskgraph/taskgraph"
> > +	"github.com/taskgraph/taskgraph/controller"
> > +	"github.com/taskgraph/taskgraph/framework"
> > +)
> > +
> > +type MapreduceBootstrapController struct {
>
> The name is very confusing and long..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskgraph/taskgraph/pull/109/files#r29827319>.
>